src/main/java/org/apache/sysml/hops/FunctionOp.java

```diff
@@ -213,10 +227,32 @@ else if ( getFunctionName().equalsIgnoreCase("svd")) {
 		}
 	}
 	
-	@Override
+	//@Override
 	protected long[] inferOutputCharacteristics( MemoTable memo )
 	{
-		throw new RuntimeException("Invalid call of inferOutputCharacteristics in FunctionOp.");
+		long[] ret = null;
+
+		//get statistics of main input
+		for(int i = 0; i < getInput().size(); i++) {
+			MatrixCharacteristics mc = memo.getAllInputStats(getInput().get(i));
+
+			if( mc.dimsKnown() ) {
+				long[] ret1 = new long[]{mc.getRows(), mc.getCols(), -1};
+
+				if ( (ret1[0] > ret[0] && ret1[1] > ret[1]) ||
+						(ret1[0] > ret[0] && ret1[1] < ret[1]) ||
+						(ret1[0] < ret[0] && ret1[1] > ret[1]) ) {
```

Comment:
I really don't understand these comparisons, but they are guaranteed to fail because ret is null.